src/app/static/src/app/components/adr/useADR.ts

```diff
@@ -0,0 +1,34 @@
+import {useStore} from "vuex";
+import {computed, onBeforeMount, onMounted} from "vue";
+
+export const useADR = () => {
+    const store = useStore();
+
+    const isGuest = computed<boolean>(() => {
+        return store.getters["isGuest"]
+    });
+    const ssoLogin = computed<boolean>(() => {
+        return store.state.adr.ssoLogin
+    });
+    const key = computed<string | null>(() => {
+        return store.state.adr.key
+    });
+
+    onBeforeMount(() => {
+        if (!isGuest.value && !ssoLogin.value) {
+            store.dispatch("adr/fetchKey").then();
```

Comment:
This was just to appease intellij warning "Promise returned from dispatch is ignored" and doesn't actually do anything. But looks like I can achieve the same thing, more explicitly by putting a void at the front. https://stackoverflow.com/questions/41278900/intellij-idea-warning-promise-returned-is-ignored-with-aysnc-await

That sound right?

Reply:
hmmm i guess that turning that arrow function into async and putting an await in front of the `store.dispatch` might please intellij, cos this is very normal, it might be freaking out cos the function is synchronous

Reply:
Ah yep that works too!